src/main/java/com/gregtechceu/gtceu/integration/kjs/recipe/GTRecipeSchema.java

```diff
@@ -382,6 +368,15 @@ public GTRecipeJS itemInputsRanged(TagPrefix orePrefix, Material material, int m
             return inputItemsRanged(ChemicalHelper.get(orePrefix, material), min, max);
         }
 
+        public GTRecipeJS NBTPredicateInput(ItemStack itemStack, NBTPredicate predicate) {
+            if (itemStack.isEmpty()) {
+                throw new RecipeExceptionJS(String.format("Input items is empty, id: %s", id));
+            }
+            gatherMaterialInfoFromStacks(itemStack);
+
+            return input(ItemRecipeCapability.CAP, new NBTPredicateIngredient(itemStack, predicate));
```

Comment:
```suggestion
            return itemInputs(InputItem.of(new NBTPredicateIngredient(itemStack, predicate), itemStack.getCount()));
```

Reply:
and with an `InputItem` named `input` as the first parameter:

```suggestion
            return itemInputs(InputItem.of(new NBTPredicateIngredient(input.ingredient, predicate), input.count));
```a